platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/PairedStreams.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.benchmark.reconnect;
+
+import com.swirlds.common.io.streams.MerkleDataInputStream;
+import com.swirlds.common.io.streams.MerkleDataOutputStream;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+/**
+ * Utility class for generating paired streams for synchronization tests.
+ */
+public class PairedStreams implements AutoCloseable {
+
+    protected BufferedOutputStream teacherOutputBuffer;
+    protected MerkleDataOutputStream teacherOutput;
+
+    protected BufferedInputStream teacherInputBuffer;
+    protected MerkleDataInputStream teacherInput;
+
+    protected BufferedOutputStream learnerOutputBuffer;
+    protected MerkleDataOutputStream learnerOutput;
+    protected BufferedInputStream learnerInputBuffer;
+    protected MerkleDataInputStream learnerInput;
+
+    protected Socket teacherSocket;
+    protected Socket learnerSocket;
+    protected ServerSocket server;
+
+    public PairedStreams() throws IOException {
+
+        server = new ServerSocket(0);
```

Comment:
Platform socket configuration should be replicated here. See `com.swirlds.platform.network.SocketConfig`

Reply:
Can `TcpFactory` be reused here?

Reply:
Not sure how critical this is given that the test runs on 127.0.0.1, but I did that.